### PR TITLE
Generated repos file

### PR DESCRIPTION
**Pull request by nuclearsandwich**
The repos file in this repository was initially created with rosinstall_generator.
We've manually updated it after changes in ROS 2 and in Space ROS-specific branches and projects.

I'd like to get to a point where we have specified via scripted logic the update process from ROS 2 and the incorporation of Space ROS modifications so that we can use the repos file as an input artifact to other builds in such a way that changes to upstream or Space ROS will generate an updated repos file sufficient to break cache on a docker build with the repos file as an input.

This is draft work for now. I usually like to work on generators until they generate 1:1 the existing file but given changes in upstream that may not be entirely achievable.

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
For the packages in spaceros-pkgs.txt, I tried to choose a "minimal" set of packages that would grab everything we wanted via dependencies. I think we could drop away even more, like the ament packages with the possible exception of ament_lint_common which we directly modify and so should depend on (not that I really expect upstream to drop it).

I think ideally the spaceros_pkgs.txt includes exactly the "feature" packages we want in Space ROS "core", just like an official variant metapackage would contain. So that as dependencies evolve underneath that's hidden from the packages list but will be reviewable when it causes changes in the generated repositories.

**Comment by nuclearsandwich** [comment]
The two scripts `generate-repos.sh` and `merge-repos.rb` combined with the current `spaceros-pkgs.txt` and `excluded-pkgs.txt` are sufficient to regenerate the current repositories file with only a few differences noted inline.

File: ros2.repos [hunk 1]
```diff
@@ -10,6 +11,7 @@ repositories:
   ament_cobra:
     type: git
     url: https://github.com/ament/ament_cobra.git
+    version: main
```
Comment (nuclearsandwich) [inline ros2.repos:14]: The lack of a `version` field is a benign error in the original repos file.

File: ros2.repos [hunk 2]
```diff
@@ -69,9 +71,6 @@ repositories:
   launch:
     type: git
     url: https://github.com/ros2/launch.git
-    # Avoid an issue that breaks the MoveIt2 build: https://github.com/ros2/launch/issues/588
-    # TODO: Revert to 'master' when this issue is fixed
-    #version: master
```
Comment (nuclearsandwich) [inline ros2.repos:74]: It's not possible with either the Ruby standard lib YAML parser (which is currently used) or PyYAML (a likely alternative for further work) to preserve comments in YAML documents. This comment has been "moved" to the human edited `spaceros.repos` file which contains overrides to be applied to the repos file generated by `rosinstall_generator` so it is preserved at the edit location even though it is not preserved here.

File: ros2.repos [hunk 3]
```diff
@@ -240,7 +239,7 @@ repositories:
   urdfdom:
     type: git
     url: https://github.com/ros/urdfdom.git
-    version: ros2
+    version: master
```
Comment (nuclearsandwich) [inline ros2.repos:242]: This change is the result of an upstream update.

**Comment by mjeronimo** [comment]
> For the packages in spaceros-pkgs.txt, I tried to choose a "minimal" set of packages that would grab everything we wanted via dependencies. I think we could drop away even more, like the ament packages with the possible exception of ament_lint_common which we directly modify and so should depend on (not that I really expect upstream to drop it).
> 
> I think ideally the spaceros_pkgs.txt includes exactly the "feature" packages we want in Space ROS "core", just like an official variant metapackage would contain. So that as dependencies evolve underneath that's hidden from the packages list but will be reviewable when it causes changes in the generated repositories.

Yes, I think that having a minimal set of 'feature' packages would be the way to go. As you say, we wouldn't have to track the packages underneath and it would give a clearer picture of the Space ROS packages.  Thumbs up from me.

**Comment by nuclearsandwich** [comment]
> Thumbs up from me.

Treating this as an approval to proceed so I can iterate on some of the proposed follow-up like separating packages in Space ROS from support packages and rebasing to ROS 2 Humble.

**Comment by mjeronimo** [comment]
Yes, please do.

**Pull request merged**